stripe/src/main/java/com/stripe/android/net/StripeApiHandler.java

```diff
@@ -419,11 +477,15 @@ static String createQuery(Map<String, Object> params)
         JSONObject headerMappingObject = new JSONObject(propertyMap);
         headers.put("X-Stripe-Client-User-Agent", headerMappingObject.toString());
 
-        if (apiVersion != null) {
-            headers.put("Stripe-Version", apiVersion);
+        if (options != null && options.getApiVersion() != null) {
+            headers.put("Stripe-Version", options.getApiVersion());
         }
 
-        if (options.getIdempotencyKey() != null) {
+        if (options != null && options.getGuid() != null) {
+            headers.put("guid", options.getGuid());
```

Comment:
Ah, I just noticed that this is for the API, not an m.stripe.com request. The API won't do anything with this cookie. The `setRequestProperty` that you used to have should be `setRequestProperty("Cookie", "m=" + options.getGuid())` down where you send the request to m.stripe.com